frontend/src/features/housingCompany/HousingCompanyCreatePage.tsx

```diff
@@ -0,0 +1,225 @@
+import React from "react";
+
+import {Button, Fieldset, IconSaveDisketteFill} from "hds-react";
+import {useImmer} from "use-immer";
+
+import {
+    useCreateHousingCompanyMutation,
+    useGetBuildingTypesQuery,
+    useGetDevelopersQuery,
+    useGetFinancingMethodsQuery,
+    useGetPropertyManagersQuery,
+} from "../../app/services";
+import {FormInputField} from "../../common/components";
+import {HousingCompanyStates, ICode, IHousingCompanyWritable, IPropertyManager} from "../../common/models";
+import {validateBusinessId} from "../../common/utils";
+
+const HousingCompanyCreatePage = (): JSX.Element => {
+    const [formData, setFormData] = useImmer<IHousingCompanyWritable>({
+        acquisition_price: {initial: null, realized: null},
+        address: {
+            postal_code: "",
+            street_address: "",
+        },
+        building_type: {id: ""},
+        business_id: "",
+        developer: {id: ""},
+        financing_method: {id: ""},
+        name: {
+            display: "",
+            official: "",
+        },
+        notes: "",
+        primary_loan: null,
+        property_manager: {id: ""},
+        state: "not_ready",
+        sales_price_catalogue_confirmation_date: "",
+    });
+    const [createHousingCompany, {error}] = useCreateHousingCompanyMutation();
+
+    const handleSaveButtonClicked = () => {
+        createHousingCompany(formData);
```

Comment:
Would be nice if the save button gave a response to the user upon success. E.g. redirecting to the housing company list page or emptying the fields - along with a comment about the success.

Reply:
As this is probably also an issue when it comes to creating apartments (and anything else that might be created), I made a ticket covering both cases: https://helsinkisolutionoffice.atlassian.net/browse/HT-149